### PR TITLE
feat: warn if link is a duplicate (issue-27)

### PR DESCRIPTION
**Pull request by brennan3**
https://github.com/delphidigital/delphi-tg-bots/issues/27

### CODE REVIEW

File: bots/reads.ts [hunk 1]
```diff
@@ -392,7 +420,14 @@ const handleUpdateUrl = async (url: string, ctx: ReadsContext, config: ReadsConf
 
   try {
     metadata = await fetchUrlMetadata(cleanUrl, config);
+    await readLinkRecentlyAdded(cleanUrl, config);
   } catch (e) {
+    if (e.message === ERROR_DUPLICATE_READ) {
+      await ctx.reply('Oops, this url was recently added already');
+      await handleNew(ctx);
```
Comment (stocks29) [inline bots/reads.ts]: in the above handling of a duplicate, we're doing a `resetState(ctx)`. should it be the same here?

File: bots/reads.ts [hunk 2]
```diff
@@ -168,6 +170,26 @@ const delphiApiUrl = (path: string, config: ReadsConfig) => {
...
+    const readsUrl = delphiApiUrl(`/api/v1/lists/${config.delphiApi.readingListId}/items?page=1&limit=50`, config);
+    const response = await fetch(readsUrl, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
+    const json = await response.json();
+    const matches = json.data.filter(read => read.link === link);
+    if (matches?.length) {
```
Comment (stocks29) [inline bots/reads.ts]: nitpick, but pretty sure `matches` will always be a defined array here

File: bots/reads.ts [hunk 3]
```diff
@@ -168,6 +170,26 @@ const delphiApiUrl = (path: string, config: ReadsConfig) => {
...
+      throw new Error(ERROR_DUPLICATE_READ);
+    }
+  } catch (e) {
+    console.error('Error determining if link was recently added: ', e);
+    throw new Error(e.message || ERROR_UNKNOWN);
```
Comment (stocks29) [inline bots/reads.ts]: better to error you think?

File: bots/reads.ts (hunk 1 above)
Comment (brennan3) [reply bots/reads.ts]: ah, prob should be consistent - this was consistent with a few lines down where we re-prompt on a new line to fetch the url - might make more sense to call handleNew() in other catch

**Review by stocks29 — APPROVED** [review]

**Pull request merged**